packages/sanity/src/core/studio/packageVersionStatus/checkForLatestVersions.ts

```diff
@@ -0,0 +1,57 @@
+import {from, map, type Observable, ReplaySubject, timer} from 'rxjs'
+import {share, switchMap} from 'rxjs/operators'
+
+//object like {sanity: '3.40.1'}
+interface VersionMap {
+  [key: string]: string | undefined
+}
+
+// How often to check for a version
+const REFRESH_INTERVAL = 1000 * 60 * 30 // every half hour
+
+//reset the observable when it completes or when it has no refcount
+const RESET_TIMER = timer(REFRESH_INTERVAL)
+
+const MODULES_URL_VERSION = 'v1'
+
+const MODULES_URL = `https://sanity-cdn.com/${MODULES_URL_VERSION}/modules/`
```

Comment:
Could we somehow share this with logic in CLI here? https://github.com/sanity-io/sanity/blob/next/packages/sanity/src/_internal/cli/util/getAutoUpdatesImportMap.ts#L11 

Maybe we just need to lift this out in some constant?

Reply:
I am not sure if there is, might be worth asking internally but for now okay with copy pasting as well